florist/api/client.py

```diff
@@ -67,3 +69,31 @@ def start(server_address: str, client: str, data_path: str, redis_host: str, red
 
     except Exception as ex:
         return JSONResponse({"error": str(ex)}, status_code=500)
+
+
+@app.get("/api/client/check_status")
+def check_status(client_uuid: str, redis_host: str, redis_port: str) -> JSONResponse:
+    """
+    Retrieve value at key client_uuid in redis if it exists.
+
+    :param client_uuid: (str) the uuid of the client to fetch from redis.
+    :param redis_host: (str) the host name for the Redis instance for metrics reporting.
+    :param redis_port: (str) the port for the Redis instance for metrics reporting.
+
+    :return: (JSONResponse) If successful, returns 200 with JSON containing the val at `client_uuid`.
+        If not successful, returns the appropriate error code with a JSON with the format below:
+            {"error": <error message>}
+    """
+    try:
+        redis_connection = redis.Redis(host=redis_host, port=redis_port)
+
+        result = redis_connection.get(client_uuid)
+
+        if result is not None:
+            assert isinstance(result, bytes)
+            return JSONResponse(json.loads(result))
+
+        return JSONResponse({"error": f"Client {client_uuid} Not Found"}, status_code=404)
+
+    except Exception as ex:
+        return JSONResponse({"error": str(ex)}, status_code=500)
```

Comment:
Small change: also log the exception with `LOGGER.exception(ex)`. You'll need to get the logger from uvicorn the same way it's done in the server.